### PR TITLE
Bump max result window to 30000

### LINKED ISSUE #19794: Bump search max result window (to 30000
For performance/scalability reasons, we set max_result_window in elasticsearch to 25000 - if there more results for a given query, we don't allow returning them.

25000 was chosen back in the day because it was above the total number of public extensions available, but we're now past that number. Let's try bumping it to 30000, hopefully that should still work and allow us to see all public extensions through the API.

The setting is `ES_MAX_RESULT_WINDOW`.

### PR DESCRIPTION
**Pull request by diox**
Fixes #19794

### CODE REVIEW

File: src/olympia/api/tests/test_pagination.py [hunk 1]
```diff
@@ -54,36 +57,43 @@ def test_metadata_with_default_page_size(self):
...
+        # Request the last page that still has a `next`.
+        page_size = 5
+        page = int(settings.ES_MAX_RESULT_WINDOW / page_size) - 1
+        request = self.factory.get('/', {'page_size': page_size, 'page': page})
         response = view(request)
         assert response.data == {
-            'page_size': 5,
-            'page_count': 5000,
+            'page_size': page_size,
+            'page_count': int(ceil(settings.ES_MAX_RESULT_WINDOW / page_size)),
```
Comment (eviljeff) [inline src/olympia/api/tests/test_pagination.py:77]: the discrepancy here makes me nervous - there's an extra `ceil` - shouldn't it be just `page + 1`?

File: src/olympia/api/tests/test_pagination.py [hunk 2]
```diff
@@ -54,36 +57,43 @@ def test_metadata_with_default_page_size(self):
...
-            'count': 30000,
+            'previous': f'http://testserver/?page={page - 1}&page_size={page_size}',
+            'next': f'http://testserver/?page={page + 1}&page_size={page_size}',
+            'count': total,
         }
 
-        request = self.factory.get('/', {'page_size': 5, 'page': 5000})
+        # Request the page that doesn't have a `next` because it's over the
+        # max result window.
+        page = int(settings.ES_MAX_RESULT_WINDOW / page_size)
```
Comment (eviljeff) [inline src/olympia/api/tests/test_pagination.py:86]:
```suggestion
        page = page + 1
```
?

File: src/olympia/api/tests/test_pagination.py [hunk 3]
```diff
@@ -54,36 +57,43 @@ def test_metadata_with_default_page_size(self):
...
+        request = self.factory.get('/', {'page_size': page_size, 'page': page})
         response = view(request)
         assert response.data == {
-            'page_size': 5,
-            'page_count': 5000,
+            'page_size': page_size,
+            'page_count': int(ceil(settings.ES_MAX_RESULT_WINDOW / page_size)),
```
Comment (eviljeff) [inline src/olympia/api/tests/test_pagination.py:91]: this should be just `'page_count': page`?

File: src/olympia/api/tests/test_pagination.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/api/tests/test_pagination.py:77]: Or if there's some case I'm missing where there's a difference, we should have a test for that scenario?

**Review by eviljeff — APPROVED** [review]
r+wc just nits with the tests

File: src/olympia/api/tests/test_pagination.py (hunk 1 above)
Comment (diox) [reply src/olympia/api/tests/test_pagination.py:77]: It was just to make the test generic and copy/pasteable

File: src/olympia/api/tests/test_pagination.py (hunk 2 above)
Comment (diox) [reply src/olympia/api/tests/test_pagination.py:86]: 89ee89001f692455fd71a4c50744afe52be2d1fc

File: src/olympia/api/tests/test_pagination.py (hunk 3 above)
Comment (diox) [reply src/olympia/api/tests/test_pagination.py]: 89ee89001f692455fd71a4c50744afe52be2d1fc

File: src/olympia/api/tests/test_pagination.py (hunk 1 above)
Comment (diox) [reply src/olympia/api/tests/test_pagination.py]: committed another simplification in 89ee89001f692455fd71a4c50744afe52be2d1fc

**Pull request merged**